src/components/organisms/LocalReplace.tsx

```diff
@@ -0,0 +1,448 @@
+        case 'Enter':
+          event.preventDefault()
+          event.stopPropagation()
+          if (event.ctrlKey && event.shiftKey) {
+            addNewlineToReplaceValue()
+          } else if (!event.repeat) {
+            // todo: [komediruzecki-13/02/2021] If we use repeat debounce should be used
+            //  could not get debounce to work with onKeyDown properly (need to debounce keyDown
+            //  but also prevent propagation on custom functionality)
+            //  original issues is that with allowing repeat and holding enter key
+            //  it would replace items fast and few seconds after holding the key it would start
+            //  entering newlines in replace text area instead of replacing following items
+            //  (probably event is fired on some re-render and is able to enter newline in text area
+            //  without this handler which stops propagation and uses different 'Enter' key logic)
+            onReplaceCurrentItem()
+          }
+          break
+        case 'Tab':
+          if (event.shiftKey && onFocusSearchInput) {
+            // Focus on search
+            event.preventDefault()
+            event.stopPropagation()
+            onFocusSearchInput()
+          }
+          break
+        case 'e':
+        case 'E':
+          if (event.altKey) {
+            event.preventDefault()
+            event.stopPropagation()
+            toggleCaseSensitiveReplace()
+          }
+          break
+        case 'F3':
+          if (event.shiftKey) {
+            navigateToNext('previous')
+            focusReplaceTextAreaInput(replaceQuery.length)
+          } else {
+            navigateToNext('next')
+            focusReplaceTextAreaInput(replaceQuery.length)
+          }
+          break
+        case 'ArrowDown':
+          if (getNumberOfTextAreaRows <= 1) {
+            event.preventDefault()
+            event.stopPropagation()
+            navigateToNext('next')
+            focusReplaceTextAreaInput(replaceQuery.length)
+          }
+          break
+        case 'ArrowUp':
+          if (getNumberOfTextAreaRows <= 1) {
+            event.preventDefault()
+            event.stopPropagation()
+            navigateToNext('previous')
+            focusReplaceTextAreaInput(replaceQuery.length)
+          }
+          break
+        case 'Escape':
+          handleOnReplaceClose()
+          break
+        case 'f':
+        case 'F':
+          if (event.ctrlKey && onReplaceToggle) {
+            onReplaceToggle(false)
+          }
+          break
+      }
+    },
+    [
+      addNewlineToReplaceValue,
+      focusReplaceTextAreaInput,
+      handleOnReplaceClose,
+      navigateToNext,
+      onFocusSearchInput,
+      onReplaceCurrentItem,
+      onReplaceToggle,
+      replaceQuery.length,
+      getNumberOfTextAreaRows,
+      toggleCaseSensitiveReplace,
+    ]
+  )
+
+  useEffect(() => {
+    if (focusingReplace) {
+      // focusReplaceTextAreaInput(replaceQuery.length)
```

Comment:
This was refactored like this:

Key from component is removed, useEffectOnce replaced with useEffect
and we are not focusing replace at end rather just focusing (previous would actually mess with changing the text area input at some different position, or when multiline in some other than last character position - it would focus to end wherever you were in text area input)

Now it uses the effect to update focus on focusing replace change (no need for key) but does not update to end of the input when replaceQuery changes

Reply:
Don't know what you want actually archive or not and which one is better

Reply:
I wanted to properly address refocusing of replace dialog,
I refactored it like we talked it seems to be working very nice.

You can check the code in the latest commit, and comment there if you find any issues.
I'll robustly test this tomorrow, but tested general use cases and all of them works nicely.